File: calendar-syncer-desktop/src/types/CalendarEvent.ts
/**
 * Calendar Event interface based on the database schema
 */
export interface CalendarEvent {
  eventID: string;
  calendarID: string;
  summary: string;
  description: string;
  startTime: Date;
  endTime: Date;
  createdTime: Date;
  updatedTime: Date;
  location?: string;
  status: string;
  organizerEmail?: string;
  attendees?: string;
  recurrence?: string;
}

/**
 * Database connection status interface
 */
export interface ConnectionStatus {
  database: boolean;
  udpListener: boolean;
}

/**
 * Application state interface
 */
export interface AppState {
  events: CalendarEvent[];
  connectionStatus: ConnectionStatus;
  loading: boolean;
  error: string | null;
}

File: CalendarDesktop/Models/CalendarEvent.cs
using System;

namespace CalendarDesktop.Models
{
    /// <summary>
    /// Represents a calendar event.
    /// </summary>
    public class CalendarEvent
    {
        public string EventID { get; set; }
        public string CalendarID { get; set; }
        public string Summary { get; set; }
        public string Description { get; set; }
        public DateTime? StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public DateTime? CreatedTime { get; set; }
        public DateTime? UpdatedTime { get; set; }
        public string? Location { get; set; }
        public string? Status { get; set; }
        public string? OrganizerEmail { get; set; }
        public string? Attendees { get; set; }
        public string? Recurrence { get; set; }
    }
}